File: TrinitySQL.Blueprint/Server/TestInfrastructure/TestDatabase.cs
using TrinitySQL.Server.Data;
using TrinitySQL.Server.Core.SharedKernel;

namespace TrinitySQL.Blueprint.Server.TestInfrastructure;

public class TestDatabase : IDisposable
{
    private readonly string _dbPath;

    public TestDatabase(TheaterSalesContext context, IQueryDispatcher queryDispatcher, string dbPath)
    {
        Context = context;
        QueryDispatcher = queryDispatcher;
        _dbPath = dbPath;
    }

    public TheaterSalesContext Context { get; }
    public IQueryDispatcher QueryDispatcher { get; }

    public void Dispose()
    {
        Context?.Dispose();
        if (File.Exists(_dbPath))
        {
            File.Delete(_dbPath);
        }
    }
}

File: TrinitySQL.Blueprint/Device/TestInfrastructure/TestDatabase.ts
import {TheaterSalesContext} from '@device/Data'
import * as fs from 'fs'

export class TestDatabase
{
  constructor(
    private readonly context: TheaterSalesContext,
    private readonly dbPath: string
  ) {}
  
  get Context(): TheaterSalesContext
  {
    return this.context
  }
  
  dispose(): void
  {
    this.context.close()
    this.deleteDatabase()
  }
  
  private deleteDatabase(): void
  {
    if (fs.existsSync(this.dbPath))
    {
      fs.unlinkSync(this.dbPath)
    }
  }
}